app/assets/javascripts/discourse/app/components/composer/toolbar-buttons.gjs

```diff
@@ -0,0 +1,68 @@
+import Component from "@glimmer/component";
+import { fn, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import { eq } from "truth-helpers";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import icon from "discourse/helpers/d-icon";
+import ToolbarPopupMenuOptions from "select-kit/components/toolbar-popup-menu-options";
+
+export default class ComposerToolbarButtons extends Component {
+  firstButton = this.args.data.groups.mapBy("buttons").flat().firstObject;
```

Comment:
`.mapBy` and `.firstObject` are Ember array prototype extensions which are [deprecated](https://deprecations.emberjs.com/id/deprecate-array-prototype-extensions).

We should avoid introducing new code using them. 

is this code supposed to be reactive? `firstButton` won't be updated when/if `args.data` is updated.

Reply:
For ref. this was all moved from

https://github.com/discourse/discourse/blob/9e4a08d08f574c85d9ba88b2e9ba65c946ab1b14/app/assets/javascripts/discourse/app/components/d-editor.gjs#L184-L187

I don't think the toolbar should changed that much but I'll move it to a getter, tks!